src/app/vertextool/qgsvertextool.cpp

```diff
@@ -2696,7 +2696,8 @@ void QgsVertexTool::setHighlightedVertices( const QList<Vertex> &listVertices, H
 
     QgsVertexMarker *marker = new QgsVertexMarker( canvas() );
     marker->setIconType( QgsVertexMarker::ICON_CIRCLE );
-    marker->setIconSize( QgsGuiUtils::scaleIconSize( 10 ) );
+    marker->setZValue( marker->zValue() + 1 );
+    marker->setIconSize( QgsGuiUtils::scaleIconSize( 12 ) );
```

Comment:
I'd drop this change, given the issues you describe with seeing the unhighlighted point marker below the marker anyway. The z value fix looks good on its own.

Reply:
Sounds good. I reverted it. If in the future I can figure out how to fix the rendering do you think it makes sense to increase the size?

Reply:
Yes, I like that idea!